test/gtest/uct/uct_test.h

```diff
@@ -117,6 +117,11 @@ class uct_test : public testing::TestWithParam<const resource*>,
     uct_test();
     virtual ~uct_test();
 
+    virtual bool has_transport(const std::string& tl_name) const;
+
+    /* Call this wrapper right before uct_config_modify */
+    virtual std::string wrap_config_str(const std::string& str) const;
```

Comment:
IMO this approach adds complexity without significant benefit. Once you need to add a new wrapper (1 simple thing), you need to control 2 functions (2 hops in the code to understand logic) instead of 1 which is simple enough.